### ISSUE TITLE
Uploading results throws exception: "Impossible to upload: Wrong API key?"

### ISSUE THREAD

**Issue by neph1** [deeplearning4j/gym-java-client#12]
#### Issue Description

I've been trying to upload the results from my evalution, I get the following exception:
"Impossible to upload: Wrong API key?"

Code used:
Client<Box, Integer, DiscreteSpace> client = ClientFactory.build("CartPole-v0", false);
client.upload(path, apiKey);

I've since used python to upload without any issues, so the path and API key are correct.

One observation I made when looking at the code (I use the compiled library, though) was that all POSTs use the following syntax:
ClientUtils.post(url + ENVS_ROOT + instanceId + XXXXX, json);

except for UPLOAD which uses:
ClientUtils.post(url + V1_ROOT + instanceId + UPLOAD, json);

In other words, UPLOAD isn't using the "/envs/" path. I don't know the code well enough to say if that is the issue or not, it's just a reflection.

I decided to try it out myself, but unfortunately, but I get many compile errors when using the latest java-gym-client repository code and dl4j 0.8.0.

#### Version Information
dl4j 0.8.0 and java-gym-client 0.8.0

**Comment by saudet** [comment]
It would be great if you could try it out with 0.8.1-SNAPSHOT (so you can use the master branch of gym-java-client for builds) and send us a pull request with those fixes!

**Comment by neph1** [comment]
Is 0.8.1-SNAPSHOT available anywhere? Can't find DL4j 0.8.1 on maven central, and RL4J snapshot is empty.

**Comment by neph1** [comment]
Nm, I seem to be able to reproduce with latest gym-java-client and dl4j 0.8.0. Let's see where I end up

**Comment by saudet** [comment]
Yes, it's on central: https://oss.sonatype.org/content/repositories/snapshots/org/deeplearning4j/
So, no, you don't have to build DL4J from source.
